GHzDACs/GHzAD_v7.md

```diff
@@ -0,0 +1,437 @@
+    d(5)    sram(+3)[7..0]    multcos(1)	Q1
+    ...
+    d(1024)sram(+1022)[7..0]  multsin(511)
+    d(1025)sram(+1023)[7..0]  multcos(511)
+
+512 * ns = 1.024us max demod length
+
+
+
+## Register Write
+
+These registers control the AD functions.
+This card is always set in slave mode for daisychain initiation of AD functions (see GHzDAC board).  
+
+    l(0)    length[15..8]   set to 0  ; ( length[15..0] = 59 )
+    l(1)    length[7..0]    set to 59
+
+    d(0)    start[7..0]        Output & command function
+                               0 = off
+                               1 = register readback
+                               2 = average mode, auto start (use n=1)
+                               3 = average mode, daisychain start
+                               4 = demodulator mode, auto start (use n=1)
+                               5 = demodulator mode, daisychain start
+                               6 = set PLL of 1GHz clock with ser[23..0], no readback
+                               7 = recalibrate AD converters, no readback
+    d(1)    startdelay[7..0]   Start delay after daisychain signal, compensates for dchain delays.
+    d(2)    startdelay[15..8]      Note longer delays than for GHzDAC
+    d(3)    ser1[7..0]         8 lowest PLL bits of serial interface 
+    d(4)    ser2[7..0]         8 middle PLL bits
+    d(5)    ser3[7..0]         8 highest PLL bits
+    d(6)    spare
+
+    d(7)    n[7..0]	           n = Number of averages in average mode
+    d(8)    n[15..8]           n = Number of total events in demodulator mode
+    d(9)    bitflip[7..0]      XOR mask for bit readout	
+    d(10)   mon0[7..0]         SMA mon0 and mon1 programming, like DAC board
+    d(11)   mon1[7..0]
+    ...
+    d(58)   spare
+
+
+## Register Readback
+
+Registers are readback according to the above data fields and additional bytes given here.
+Length of readback = 46.
+
+    d(0)    build[7..0]         Build number of FPGA code.  
+    d(1)    clockmon[7..0]      bit0 = NOT(LED1 light) = no PLL in external 1GHz clock 
+                                bit1=dclkA output (phase of data stream)
+                                bit2=dclkA delayed 1ns output
+                                bit3=dclkB output
+                                bit4=dclkB delayed 1ns output
+    d(2)    trigs[7..0]         Count of number of triggers (same as DAC sramcount) 
+    d(3)    trigs[15..8]            reset with start=2,3,4,5
+    d(4)    npackets[7..0]      Counter of number of packets received (reg and SRAM)
+    d(5)    badpackets[7..0]    Number of packets received with bad CRC 
+    d(6)    spare
+    ...
+    d(45)   spare[7..0]         set to 0
+
+
+## Demodulator output
+
+After demodulation is done on each retrigger, demodulator output is put into a FIFO from channel 0 to `numchan[7..0]-1`.
+Once 11 channels are in FIFO (44 bytes), or end of all retriggering, the ethernet packet is sent by pulling bytes out of FIFO, 44 bytes per output event.
```

Comment:
11 IQ pairs -- channel is confusing because you sometimes use it to mean mixer table entries.